sensor/kubernetes/listener/resources/deployment_store.go

```diff
@@ -181,6 +210,15 @@ func (ds *DeploymentStore) Get(id string) *storage.Deployment {
 	return wrap.GetDeployment()
 }
 
+// GetBuiltDeployment returns a cloned deployment for supplied id
+func (ds *DeploymentStore) GetBuiltDeployment(id string) *storage.Deployment {
+	ds.lock.Lock()
+	defer ds.lock.Unlock()
+	// TODO(ROX-16138): return nil if the deployment is not built yet
+	wrap := ds.getWrapNoLock(id)
+	return wrap.GetDeployment().Clone()
```

Comment:
I think this will panic if `wrap` is nil. We should add a `nil` check here before returning.

Reply:
I could be wrong but I don’t think it would panic because `GetDeployment` and `Clone` check whether the caller is nil or not.

Reply:
You're right, it does. The following test passes:
```go
func (s *deploymentStoreSuite) Test_BuildDeployment() {
	d := s.deploymentStore.GetBuiltDeployment("random")
	s.Require().Nil(d)
}
```